source/_posts/2020-09-12-android-240-release.markdown

```diff
@@ -0,0 +1,70 @@
+
+The first major change in this release is that the app will no longer send needless updates to your Home Assistant server when there is nothing new to report state-wise. This helps cut down on the amount of data that gets sent out. Some attributes attached to some sensors update outside of the state, to combat this change, we have split those attributes to become their own sensor. For example, the audio sensor had an attribute to tell if headphones were currently plugged into the device. This in itself is a state change and is now a binary sensor.
+
+We have opted to keep the existing attributes where they are for this release, this way they do not break existing templates. However, they may trigger slower until you move to the new sensor. All of the new sensors are also disabled by default (with the exception of battery), so you may want to enable the ones you care for. The following sensors have had their attributes split up:
+
+- Audio Sensor - All attributes. The sensor name for new users is changed to `Ringer Mode` to better align with what the state represents.
+- Battery State Sensor - All attributes.
+- Bluetooth Connection - Only `is_bt_on` now has its own binary sensor. The other attributes update with the state.
+- Storage Sensor - External storage attributes. The new sensor will behave the same as the current storage sensor which has also now been labeled as Internal Storage.
+- WiFi Connection - All attributes with the exception of `is_hidden`.
+
+These attributes mentioned above will be removed in the next release. Please make your switch to the new sensors now to avoid the breaking change then.
+
+Attributes have also had a slight breaking change as we are now sending the correct data type for the attribute. Previously they were all sent as a string, which meant binary values needed to be wrapped in quotes.
+
+The Geocoded location sensor also had its attribute of `Location` split into the attributes they reflect as `latitude` and `longitude`.
+
+## Foreground Service Worker
+
+We have seen issues where some of the sensors do not update reliably; There were times where the sensor worker was not able to run. We have gone ahead and converted this to become a foreground service, which means from time to time, you will see a low priority notification when the sensors are updating. This low priority notification does not make a sound and just appears briefly in the status bar. We felt this to be a good compromise instead of having to use a persistent notification.
```

Comment:
```suggestion
We have seen issues where some of the sensors do not update reliably; There were times when the sensor worker could not run. We have gone ahead and converted this to become a foreground service, which means from time to time, you will see a low priority notification when the sensors are updating. This low priority notification does not make a sound and just appears briefly in the status bar. We felt this to be a good compromise instead of having to use a persistent notification.
```